File: mini-core/src/main/java/com/mini/core/security/digest/MD5.java
package com.mini.core.security.digest;

import java.nio.charset.Charset;

public final class MD5 extends BaseDigest {
	public MD5() {
		super("MD5");
	}
	
	public static MD5 getInstance() {
		return new MD5();
	}
	
	public static String encode(String data, Charset charset) {
		return getInstance().update(data, charset).encode();
	}
	
	public static String encode(String data) {
		return getInstance().update(data).encode();
	}
}

File: mini-util/src/main/kotlin/com/mini/core/security/digest/MD5.kt
@file:Suppress("unused")

package com.mini.core.security.digest

import java.nio.charset.Charset

class MD5 : BaseDigest("MD5") {
    companion object {
        @JvmStatic
        val instance: MD5
            get() = MD5()

        @JvmStatic
        fun encode(data: String, charset: Charset): String {
            return instance.update(data, charset).encode()
        }

        @JvmStatic
        fun encode(data: String): String {
            return instance.update(data).encode()
        }
    }
}